manifests/config.pp

```diff
@@ -29,12 +29,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-class openvpn::config inherits openvpn::params {
+class openvpn::config {
 
-  if $::osfamily == 'Debian' {
+  if $facts['osfamily'] == 'Debian' {
```

Comment:
@bastelfreak I have to use `$facts['osfamily']` here. `$facts['os']['family']` works inside puppet fine but i get `[] Operator on Undef is not allowed` on rspec tests.

Any idea?

Reply:
I assume because of this:
https://github.com/voxpupuli/puppet-openvpn/blob/139c72778008b91469d445f4f2c89d5c35d57ef4/spec/unit/openvpn_module_spec.rb#L15-L56

It explicitly mocks the `osfamily` hash and doesn't use rspec-puppet-facts.

Reply:
structured facts with rspec are facts[:os]['family'].
hope this help.